### PR TITLE
Show num cards

### PR DESCRIPTION
**Pull request by Aaron-Fine**
This PR implements [Issue 88](https://github.com/jak103/uno/issues/88) which was supposed to show the user the number of cards for each of the other players in a game. 

Summary of important changes:

* A new endpoint was added to the server (/cardcount/{game id}) which can be tried out via `curl localhost:8080/cardcount/5f1df2faca9a30fdd91f2dd3`, assuming the game id given already exists.
* The new endpoint was added to `unoServices`
* The game display was updated to use data from the new endpoint for displaying player names and hand counts
* the game display was formatted to resolve lint errors

NOTE: I can't test the changes to the front end right now. 

1) The database always just returns an empty gameState structure, which is needed to get player name and hand count info
2) The front end doesn't seem to be able to join games right now. This problem exists for me on both this branch and on dev so I think something is broken on dev and not here.

### CODE REVIEW

File: server/gameData.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	"github.com/jak103/uno/db"
+)
+
+// HandInfo is a JSON struct for holding summary information about each players hand for a given game.
+type HandInfo struct {
+	PlayerName string `json:"name"`
+	CardCount  int    `json:"numCards"`
+}
+
+func getCardCount(gameID string) (string, error) {
```
Comment (jak103) [inline server/gameData.go:16]: Let's not create a new file for a single function. Move this over to uno.go.

File: server/gameData_test.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetCardCount(t *testing.T) {
```
Comment (jak103) [inline server/gameData_test.go:9]: I wont' accept PRs without unit tests. Complete this test.

**Comment by logan-hendricks** [comment]
> 2. The front end doesn't seem to be able to join games right now. This problem exists for me on both this branch and on dev so I think something is broken on dev and not here.

@Aaron-Fine This should be working now if you merge the latest dev unto your branch.

File: server/gameData.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/jak103/uno/db"
+)
+
+// HandInfo is a JSON struct for holding summary information about each players hand for a given game.
+type HandInfo struct {
```
Comment (logan-hendricks) [inline server/gameData.go:11]: While this is not a database `model` per say... My opinion is that I think this should be declared in the `model` package, as we also have the `card` model located there.  Basically as a global location for data structures to be housed.

File: server/gameData_test.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+	//TODO: Setup mock database
```
Comment (logan-hendricks) [inline server/gameData_test.go:10]: Ideally, after the changes Matt made, the MockDB struct should already be preloaded to return the data you need for your unit tests and shouldn't need any `setup`. It's selected by default on unit tests. You'll just need to check validate your tests according to what it returns.

File: server/gameData.go (hunk 3 above)
Comment (Aaron-Fine) [reply server/gameData.go:11]: Done. I put it in `model/Game.go` as it it tied closely to that struct anyway.

File: server/gameData.go (hunk 1 above)
Comment (Aaron-Fine) [reply server/gameData.go:16]: Done.

File: server/gameData_test.go (hunk 4 above)
Comment (Aaron-Fine) [reply server/gameData_test.go]: I can't figure out how to get the mock database to work. 

Here is how I am setting up the database:
```
	database, err := db.GetDb()
	assert.NoError(t, err)

	game, err := database.CreateGame()
	assert.NoError(t, err)

	player, err := database.CreatePlayer("Player 1")
	assert.NoError(t, err)

	err = database.JoinGame(game.ID, player.ID)
	assert.NoError(t, err)
```

However, when I call `database.LookupGameByID(gameID)` I always get ` &{5f1e42344170d7831aeaec08  [] [] [] 0  false}` as a result (note that the UUID does change but the rest remains the same, namely, unpopulated).

I originally thought that there was something wrong with the mock, and indeed, `JoinGame` doesn't seem to do anything. My best guess as to an implementation is
```
func (db *mockDB) JoinGame(id string, username string) error {
	db.games[id].Players[0] = db.players[username]
	return nil
}
```
Yet I'm not getting anything different from the lookup, so I'm stumped and confused.

Comment (logan-hendricks) [reply server/gameData_test.go]: @Aaron-Fine I found the issue with JoinGame and pushed up changes to the MockDB.  You should be able to test this now. 

You were correct. that particular endpoint was a stub. Let me know if you face errors now.

**Comment by jak103** [comment]
Closing this as it is OBE based on the new game state.

**Pull request closed without merging**